File: VB/XPO_XtraScheduler_Simple_Example/XPResource.vb
Imports DevExpress.Xpo
Imports System.Drawing

Namespace XPO_XtraScheduler_Simple_Example

'#Region "#xpresource"
    ' XP object
    Public Class XPResource
        Inherits XPObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public ResId As Integer

        <Size(SizeAttribute.Unlimited)>
        Public Name As String ' Resource.Caption

        Public Color As Integer

        Public Image As Image
    End Class
'#End Region  ' #xpresource
End Namespace

File: CS/XPO_XtraScheduler_Simple_Example/XPResource.cs
using DevExpress.Xpo;
using System;
using System.Drawing;

namespace XPO_XtraScheduler_Simple_Example {
    #region #xpresource
    // XP object
    public class XPResource : XPObject {
        public XPResource(Session session) : base(session) { }
        public int ResId;
        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Name;              // Resource.Caption
        public Int32 Color;
        public Image Image;
    }
    #endregion #xpresource
}